lib/sig.js

```diff
@@ -25,7 +25,8 @@ exports.getTokens = function(html5playerfile, options, callback) {
   if (cachedTokens) {
     callback(null, cachedTokens);
   } else {
-    html5playerfile = 'http:' + html5playerfile;
+    if (html5playerfile.lastIndexOf('/', 0) === 0 && html5playerfile.lastIndexOf('//', 0) !== 0) html5playerfile = 'https://www.youtube.com' + html5playerfile; //youtube forces https and www. otherwise we get a 301
```

Comment:
Using `lastIndexOf` for this is unintuitive. Something like this might be more clear:

```javascript
  ...
  if (cachedTokens) {
    callback(null, cachedTokens);
  } else {
    // html5playerfile might be relative like "/yts/jsbin/player-en_US-vflV3n15C/base.js"
    if (html5playerfile[0] === '/') { // no protocol - add https:
      if (html5playerfile[1] !== '/') { // no domain - add youtube domain
        html5playerfile = '//www.youtube.com' + html5playerfile;
      }
      html5playerfile = 'https:' + html5playerfile;
    }

    var myrequest = options.request || request;
    myrequest(html5playerfile, options.requestOptions, function(err, body) {
    ...
```
and also uses https for the "old way" as well.